### ISSUE TITLE
Flow 0.36.0 with draft-js 0.9.1

### ISSUE THREAD

**Issue by rahulthewall** [facebook/draft-js#947]
I see that this is a common (or oft-encountered) issue, and I've gone through the previous issue #661 detailing the same troubles. On that issue, I've used the [example .flowconfig provided](https://github.com/facebook/draft-js/issues/661#issuecomment-248846071). 

I've given my flowconfig below:

[flowconfig.txt](https://github.com/facebook/draft-js/files/708815/flowconfig.txt)

Using the above configuration, I ran into the following errors:

```
Launching Flow server for /Users/rahul/work/deep_impact/spectra/mango
Spawned flow server (pid=64150)
Logs will go to /private/tmp/flow/zSUserszSrahulzSworkzSdeep_impactzSspectrazSmango.log
node_modules/react-side-effect/node_modules/fbjs/flow/include/PromiseMap.js:19
 19: import type * as Promise from 'Promise';
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ import type * as Promise. This is invalid syntax. Maybe you meant: `import type Promise from "Promise"`?

node_modules/react-side-effect/node_modules/fbjs/flow/include/PromiseMap.js:27
 27:   _deferred: {[key:string]: Deferred};
                                 ^^^^^^^^ Deferred. Application of polymorphic type needs <list of 2 arguments>. (Can use `*` for inferrable ones)

node_modules/react-side-effect/node_modules/fbjs/flow/include/PromiseMap.js:51
 51:   entries: {[key: string]: Deferred},
                                ^^^^^^^^ Deferred. Application of polymorphic type needs <list of 2 arguments>. (Can use `*` for inferrable ones)

node_modules/react-side-effect/node_modules/fbjs/flow/include/PromiseMap.js:53
 53: ): Deferred {
        ^^^^^^^^ Deferred. Application of polymorphic type needs <list of 2 arguments>. (Can use `*` for inferrable ones)

node_modules/react-side-effect/node_modules/fbjs/lib/CSSCore.js:0
CSSCore. Duplicate module provider
current provider. See: node_modules/react-side-effect/node_modules/fbjs/flow/include/CSSCore.js:0

node_modules/react-side-effect/node_modules/fbjs/lib/Deferred.js:0
Deferred. Duplicate module provider
current provider. See: node_modules/react-side-effect/node_modules/fbjs/flow/include/Deferred.js:0

... (and so on)
```

I then came across a similar issue on the flowtype repo: https://github.com/flowtype/flow-typed/issues/586. A suggested fix was to nuke the `node_modules/react-side-effect/node_modules/fbjs` root and run flow again. I tried that, and then I ran into the following issues:

```
Launching Flow server for /Users/rahul/work/deep_impact/spectra/mango
Spawned flow server (pid=65566)
Logs will go to /private/tmp/flow/zSUserszSrahulzSworkzSdeep_impactzSspectrazSmango.log
src/clients//views/ArticleEditor/ArticleEditor.js:163
163: export default withStyles(s)(connect(mapStateToProps, mapDispatchToProps)(init(initCfg)(ArticleEditor))); // eslint-disable-line max-len
                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ function call. Function cannot be called on any member of intersection type
163: export default withStyles(s)(connect(mapStateToProps, mapDispatchToProps)(init(initCfg)(ArticleEditor))); // eslint-disable-line max-len
                                  ^^^^^^^ intersection
  Member 1:
   50:   declare function connect<A, OP>(
                                 ^ polymorphic type: function type. See lib: flow-typed/npm/react-redux_v4.x.x.js:50
  Error:
  163: export default withStyles(s)(connect(mapStateToProps, mapDispatchToProps)(init(initCfg)(ArticleEditor))); // eslint-disable-line max-len
                                            ^^^^^^^^^^^^^^^ function. This type is incompatible with
   51:     ...rest: Array<void> // <= workaround for https://github.com/facebook/flow/issues/2360
                          ^^^^ undefined. See lib: flow-typed/npm/react-redux_v4.x.x.js:51
  Member 2:
   54:   declare function connect<A, OP>(
                                 ^ polymorphic type: function type. See lib: flow-typed/npm/react-redux_v4.x.x.js:54
  Error:
  163: export default withStyles(s)(connect(mapStateToProps, mapDispatchToProps)(init(initCfg)(ArticleEditor))); // eslint-disable-line max-len
                                            ^^^^^^^^^^^^^^^ function. This type is incompatible with
   55:     mapStateToProps: Null,
                            ^^^^ union: null | undefined. See lib: flow-typed/npm/react-redux_v4.x.x.js:55

... (and so on)
```

These issues were not present earlier. I am also providing my original flowconfig below:

[flowconfig_original.txt](https://github.com/facebook/draft-js/files/708827/flowconfig_original.txt)

I would be grateful if someone could point out what is the correct way to configure flow with draft-js. 

Thanks

**Comment by flarnie** [comment]
Hi @rahulthewall - thanks for reporting this. We have done some flow fixes since v0.9.1 and I'm wondering if this still repros when you use the latest versions of Draft.js and Flow? Or rather Draft.js and flow@^0.42.0, which is what we are using on Github?

Also it looks like those errors are from `fbjs`, not sure why that is coming up.

**Comment by flarnie** [comment]
Sorry that this got stale - hopefully the latest version of Draft has fixed this issue for you, and thanks for reporting it.
